src/components/DelegateForm.js

```diff
@@ -63,14 +63,16 @@ class DelegateForm extends React.Component {
     const address = this.props.address
     const validatorAddress = this.props.selectedValidator.operator_address
     let messages = []
+    const decimals = this.props.network.data.decimals
+    const d = decimals ? decimals : 1e6;
     if(this.props.redelegate){
       messages.push({
         typeUrl: "/cosmos.staking.v1beta1.MsgBeginRedelegate",
         value: {
           delegatorAddress: address,
           validatorSrcAddress: this.props.validator.operator_address,
           validatorDstAddress: validatorAddress,
-          amount: coin(parseInt(parseFloat(amount) * 1000000), this.props.network.denom),
+          amount: coin(parseInt(parseFloat(amount) * d), this.props.network.denom),
```

Comment:
amount: coin(parseInt(parseFloat(amount) * math.pow(10, d), this.props.network.denom),